db/version_set.cc

```diff
@@ -3831,16 +3832,27 @@ Status VersionSet::ProcessManifestWrites(
         }
       }
       if (version == nullptr) {
-        version = new Version(last_writer->cfd, this, file_options_,
-                              last_writer->mutable_cf_options, io_tracer_,
-                              current_version_number_++);
-        versions.push_back(version);
-        mutable_cf_options_ptrs.push_back(&last_writer->mutable_cf_options);
-        builder_guards.emplace_back(
-            new BaseReferencedVersionBuilder(last_writer->cfd));
-        builder = builder_guards.back()->version_builder();
-      }
-      assert(builder != nullptr);  // make checker happy
+        bool is_all_wal_maniputations = true;
+        for (const auto& e : last_writer->edit_list) {
+          if (!e->IsWalManipulation()) {
+            is_all_wal_maniputations = false;
+            break;
+          }
+        }
```

Comment:
I wonder whether this computation can be moved to the constructor of `ManifestWriter`, and stored in a new member of `ManifestWriter`.